### PR TITLE
Split off log functions into a new module

### PR DESCRIPTION
**Pull request by kalikiana**
The Utils module has grown a diverse set of features and it may be time to give the log helpers their own space.

This PR also cleans up inconsistencies in how the functions are being called in different places.

### CODE REVIEW

**Comment by kalikiana** [comment]
> +1 good idea. As you create a new module I suggest you also make sure you have 100% statement coverage for this module at least, e.g. create t/16-log.t

We already have `t/28-logging.t` but I double-checked what's missing and apparently formatting, channel lookup and removal weren't covered so far, so I added those.

**Comment by kalikiana** [comment]
Note that during another rebase I noticed that the case of a broken needle didn't seem to be covered, so I added a case for it (the UI doesn't distinguish that).

**Comment by okurz** [comment]
Please split into commits. This is definitely too much which you squish together with just extracting log functions which is a big enough change on its own already. For something like the added test as still some tests fail for you it might be easier if you even split it out into its own PR which we can merge easily with less fear to break something else and a potential need to revert.

File: t/ui/12-needle-edit.t [hunk 1]
```diff
@@ -518,10 +518,22 @@ subtest 'Showing new needles limited to the 5 most recent ones' => sub {
...
+        join("\n", 'Could not parse needle: inst-timezone-text for opensuse 13.1', @expected_needle_warnings),
+        'warning about deleted needle displayed (beside new needle warnings)'
+    );
+};
+
 subtest 'Deletion of needle is handled gracefully' => sub {
     # delete needle on disk and reload the needle editor
     $needle_json_file->remove;
-    $driver->get($driver->get_current_url);
+    $driver->refresh();
```
Comment (Martchus) [inline t/ui/12-needle-edit.t]: Changes like that should be at least in a separate commit.

Comment (kalikiana) [reply t/ui/12-needle-edit.t]: #2884

**Comment by okurz** [comment]
I found a ticket for you that matches this PR: https://progress.opensuse.org/issues/27868 :)
